lib/rpmts.cc

```diff
@@ -151,6 +151,13 @@ int rpmtsRebuildDB(rpmts ts)
 	    rc = rpmdbRebuild(ts->rootDir, NULL, NULL, rebuildflags);
 	rpmtxnEnd(txn);
     }
+    /* Re-create lock file */
+    ts->lockPath = _free(ts->lockPath);
+    ts->lock = rpmlockFree(ts->lock);
```

Comment:
After looking up the logic in rpmtxnBegin() I obviously see why these are needed, but it looks wrong, especially the lockpath free, in a function that does a database rebuild.

Just put these in an appropriately named helper function (maybe rpmtxnFree(rpmts ts) to suggest the relation to rpmtxn) and call from rpmtsFree() as well.